relayer/chainproxy/connector.go

```diff
@@ -101,16 +108,45 @@ func (connector *Connector) Close() {
 	}
 }
 
-func (connector *Connector) GetRpc(block bool) (*rpcclient.Client, error) {
+func (connector *Connector) increaseNumberOfClients(ctx context.Context) {
+	utils.LavaFormatDebug("No clients are free, increasing number of clients", nil)
+	var rpcClient *rpcclient.Client
+	var err error
+	for connectionAttempt := 0; connectionAttempt < MaximumNumberOfParallelConnectionsAttempts; connectionAttempt++ {
+		nctx, cancel := context.WithTimeout(ctx, DialTimeout)
+		rpcClient, err = rpcclient.DialContext(nctx, connector.addr)
```

Comment:
new connector creating a client and this one have a lot of dup code, create an inner function and reuse it

Reply:
not in this task.